File: Free Html To Pdf Converter for ASP.NET MVC/[C#]-Free Html To Pdf Converter for ASP.NET MVC/C#/SelectPdf.Samples/Controllers/ConvertHtmlCodeToPdfController.cs
using System;
using System.Web.Mvc;

namespace SelectPdf.Samples.Controllers
{
    public class ConvertHtmlCodeToPdfController : Controller
    {
        // GET: ConvertHtmlCodeToPdf
        public ActionResult Index()
        {
            string stringdata = "";
            
            stringdata = @"<html>
    <body>
        Hello World from selectpdf.com.
    </body>
</html>
";
            ViewData.Add("TxtHtmlCode", stringdata);
            return View();
        }
        
        [HttpPost]        
        [ValidateInput(false)]
        public ActionResult SubmitAction(FormCollection collection)
        {
            // read parameters from the webpage
            string htmlString = collection["TxtHtmlCode"];
            string baseUrl = collection["TxtBaseUrl"];

            string pdf_page_size = collection["DdlPageSize"];
            PdfPageSize pageSize = (PdfPageSize)Enum.Parse(typeof(PdfPageSize),
                pdf_page_size, true);

            string pdf_orientation = collection["DdlPageOrientation"];
            PdfPageOrientation pdfOrientation =
                (PdfPageOrientation)Enum.Parse(typeof(PdfPageOrientation),
                pdf_orientation, true);

            int webPageWidth = 1024;
            try
            {
                webPageWidth = Convert.ToInt32(collection["TxtWidth"]);
            }
            catch { }

            int webPageHeight = 0;
            try
            {
                webPageHeight = Convert.ToInt32(collection["TxtHeight"]);
            }
            catch { }

            // instantiate a html to pdf converter object
            HtmlToPdf converter = new HtmlToPdf();

            // set converter options
            converter.Options.PdfPageSize = pageSize;
            converter.Options.PdfPageOrientation = pdfOrientation;
            converter.Options.WebPageWidth = webPageWidth;
            converter.Options.WebPageHeight = webPageHeight;

            // create a new pdf document converting an url
            PdfDocument doc = converter.ConvertHtmlString(htmlString, baseUrl);

            // save pdf document
            byte[] pdf =doc.Save();

            // close pdf document
            doc.Close();

            // return resulted pdf document
            FileResult fileResult = new FileContentResult(pdf, "application/pdf");
            fileResult.FileDownloadName = "Document.pdf";
            return fileResult;

        }
    }
}

File: Free Html To Pdf Converter for ASP.NET MVC/[VB]-Free Html To Pdf Converter for ASP.NET MVC/VB/SelectPdf.Samples/Controllers/ConvertHtmlCodeToPdfController.vb
Imports SelectPdf

Namespace Controllers
    Public Class ConvertHtmlCodeToPdfController
        Inherits Controller

        ' GET: ConvertHtmlCodeToPdf
        Public Function Index() As ActionResult
            Dim stringdata As String
            stringdata = "<html>" & vbCr & vbLf & " <body>" & vbCr & vbLf &
                "  Hello World from selectpdf.com." & vbCr & vbLf & " </body>" _
                & vbCr & vbLf & "</html>" & vbCr & vbLf
            ViewData.Add("TxtHtmlCode", stringdata)
            Return View()
        End Function

        <HttpPost>
        <ValidateInput(False)>
        Public Function SubmitAction(collection As FormCollection) As ActionResult
            ' read parameters from the webpage
            Dim htmlString As String = collection("TxtHtmlCode")
            Dim baseUrl As String = collection("TxtBaseUrl")

            Dim pdf_page_size As String = collection("DdlPageSize")
            Dim pageSize As PdfPageSize = DirectCast([Enum].Parse(GetType(PdfPageSize),
                    pdf_page_size, True), PdfPageSize)

            Dim pdf_orientation As String = collection("DdlPageOrientation")
            Dim pdfOrientation As PdfPageOrientation =
                    DirectCast([Enum].Parse(GetType(PdfPageOrientation),
                    pdf_orientation, True), PdfPageOrientation)

            Dim webPageWidth As Integer = 1024
            Try
                webPageWidth = Convert.ToInt32(collection("TxtWidth"))
            Catch
            End Try

            Dim webPageHeight As Integer = 0
            Try
                webPageHeight = Convert.ToInt32(collection("TxtHeight"))
            Catch
            End Try

            ' instantiate a html to pdf converter object
            Dim converter As New HtmlToPdf()

            ' set converter options
            converter.Options.PdfPageSize = pageSize
            converter.Options.PdfPageOrientation = pdfOrientation
            converter.Options.WebPageWidth = webPageWidth
            converter.Options.WebPageHeight = webPageHeight

            ' get base url
            If baseUrl = "" Then
                baseUrl = Me.ControllerContext.HttpContext.Request.
                    Url.AbsoluteUri.Substring(
                    0, Me.ControllerContext.HttpContext.Request.
                    Url.AbsoluteUri.Length - "CurrentWebPageToPdf/SubmitAction".Length - 1)
            End If

            ' create a new pdf document converting an url
            Dim doc As PdfDocument = converter.ConvertHtmlString(htmlString, baseUrl)

            ' save pdf document
            Dim pdf As Byte() = doc.Save()

            ' close pdf document
            doc.Close()

            ' return resulted pdf document
            Dim fileResult As FileResult = New FileContentResult(pdf, "application/pdf")
            fileResult.FileDownloadName = "Document.pdf"
            Return fileResult

        End Function
    End Class
End Namespace